runtime/vm/createramclass.cpp

```diff
@@ -2901,14 +2907,15 @@ internalCreateRAMClassFromROMClassImpl(J9VMThread *vmThread, J9ClassLoader *clas
 					 * properties from the leafComponentType. A 2D array is an array of references so it can never be flattened, however, its
 					 * elements may be flattened arrays.
 					 */
-					ramArrayClass->classFlags |= (elementClass->classFlags & (J9ClassIsFlattened | J9ClassLargestAlignmentConstraintReference | J9ClassLargestAlignmentConstraintDouble));
+					ramArrayClass->classFlags |= (elementClass->classFlags & arrayFlags);
+
 				}
 				ramArrayClass->leafComponentType = leafComponentType;
 				ramArrayClass->arity = arity;
 				ramArrayClass->componentType = elementClass;
 				ramArrayClass->module = leafComponentType->module;
 
-				if (J9_IS_J9CLASS_FLATTENED(elementClass)) {
+				if (J9_IS_J9CLASS_FLATTENED(ramArrayClass)) {
```

Comment:
What's the consequence of this change? Is this a bug fix rolled into this PR?

Reply:
I think I see - previously the element determined if the array was flattened.

Reply:
Does there need to be a check for the element type being a VT?

Reply:
> previously the element determined if the array was flattened.

Yes this code was technically wrong before but there was no impact since prior to this change the element determined if the array is flattened or not. 

Since flattening or array/element will now be done independently, one must only look at the array flags.

 > Does there need to be a check for the element type being a VT?
For a singleton type, being a VT is a prerequisite of flattening. So `J9_IS_J9CLASS_FLATTENED(elementClass) == TRUE` implies VT. Arrays cannot be VTs, only singleton types, so `J9_IS_J9CLASS_FLATTENED(ramArrayClass) == TRUE` implies that elementType is flattened and also a VT.

Reply:
Short answer is I dont think there needs to be a VT check here, since its implied with the flattening check

Reply:
I wasn't clear - it looks like all arrays will get the flattened flag if the new option is set, which seems incorrect.

Reply:
Ah, `arrayFlags` is a mask - I read it as being assigned to the flags.